CONTRIBUTING.md

```diff
@@ -164,3 +176,36 @@ Do your best to follow the clean code guidelines.
 - Don’t return an error code, throw an exception instead.
 
 Ref: Book: Clean Code by Robert C. Martin (Uncle Bob)
+
+## Compilation and Installation
+This guide is helpful to compile and install Nmstate from source, for installing stable release or other installation methods, pleaser refer to [Nmstate installation guide](https://nmstate.io/user/install.md)
+
+### Dependencies
+Nmstate requires NetworkManager 1.26 or later version installed and started.
+
+In order to support specific capabilities, additional packages are required:
+
+OpenVSwitch support through the NM provider requires NetworkManager-ovs.
+
+Extended ovsdb support requires openvswitch-2.11 and python3-openvswitch2.11 or greater.
+
+Team interface support through the NM provider requires NetworkManager-team
+
+### Clone the source
+```
+git clone https://github.com/nmstate/nmstate.git
+cd nmstate
+pip install --user --upgrade .
+```
+
+### Compilation 
+```
+make
```

Comment:
I think that we need to guarantee that the system installed the official compiler for the rust programming language, https://rustup.rs/.

Reply:
I have add it

Reply:
@liangwen12year this comment is confusing. Instead of referring them to rust up, the right way to check everything is fine is running `cargo --version`.

Reply:
Sorry, rust up is specifically for the experienced developer, @elomscansio , please ignore my comment.